### DISCUSSION TITLE
INVALID_SPI

### DISCUSSION THREAD

**Discussion by chikicc** [strongswan/strongswan#919] [Q&A]
Hi,

Sometimes Strongswan (5.5.1) sends packets to a Cisco router with "invalid" ESP SPI.
Strongswan has two established IKE_SAs with installed CHILD_SAs. Both SAs are for the same connection but Cisco knows only one of them.
Strongswan sends packets using the SPI of the SA the Cisco is not aware of.

Cisco router sends INVALID_SPI notify to Strongswan and receives ACK response for it but nothing else seems to happen.
Should Strongswan somehow react to INVALID_SPI messages received from Cisco router?

**Comment by tobiasbrunner** [comment]
> Strongswan has two established IKE_SAs with installed CHILD_SAs. Both SAs are for the same connection but Cisco knows only one of them.

You might want to try to determine why that is (i.e. study the logs).

> Should Strongswan somehow react to INVALID_SPI messages received from Cisco router?

Does it send that in an unprotected INFORMATIONAL message? Or on the other IKE_SA? (Either way, strongSwan currently does not handle - or send - INVALID_SPI at all.)
